cpp/src/arrow/array/builder_nested.h

```diff
@@ -395,8 +395,17 @@ class ARROW_EXPORT StructBuilder : public ArrayBuilder {
     return Status::OK();
   }
 
-  Status AppendNull() final { return Append(false); }
+  /// \brief Append a null value. Automatically appends a null to each child
+  /// builder.
```

Comment:
Do we really want this? It seems it would create null bitmaps in the children even if all nulls are at the top level.

Reply:
We may want to add some ArrayBuilder methods `AppendEmpty` or `AppendZero`/`AppendZeros` to append arbitrary non-null empty/zero-like values.

Reply:
Ah good point, I didn't think about the triggering of creating a validity bitmap. 

AppendEmptyValue / AppendEmptyValues ("AppendZero" makes less sense for e.g. BinaryBuilder) seems like as reasonable thing to add, I'll look into this. This definitely increases the scope of the PR, though, do you think it would be better to do this as a follow up?

Reply:
As you prefer. A followup is fine to me.